pkg/cloud_provider/clientset/clientset.go

```diff
@@ -56,19 +66,40 @@ func New(kubeconfigPath string) (Interface, error) {
 		rc, err = rest.InClusterConfig()
 	}
 	if err != nil {
-		klog.Fatalf("Failed to read kubeconfig: %v", err)
+		return nil, fmt.Errorf("failed to read kubeconfig: %w", err)
 	}
 
 	clientset, err := kubernetes.NewForConfig(rc)
 	if err != nil {
-		klog.Fatal("failed to configure k8s client")
+		return nil, fmt.Errorf("failed to configure k8s client: %w", err)
 	}
 
 	return &Clientset{k8sClients: clientset}, nil
 }
 
-func (c *Clientset) GetPod(ctx context.Context, namespace, name string) (*corev1.Pod, error) {
-	return c.k8sClients.CoreV1().Pods(namespace).Get(ctx, name, metav1.GetOptions{})
+func (c *Clientset) ConfigurePodLister(nodeName string) {
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(
+		c.k8sClients,
+		resyncDuration,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = "spec.nodeName=" + nodeName
+		}),
+	)
+	podLister := informerFactory.Core().V1().Pods().Lister()
+
+	ctx := context.Background()
+	informerFactory.Start(ctx.Done())
+	informerFactory.WaitForCacheSync(ctx.Done())
+
+	c.podLister = podLister
+}
+
+func (c *Clientset) GetPod(namespace, name string) (*corev1.Pod, error) {
+	if c.podLister == nil {
+		return nil, errors.New("GetPod is called before ConfigurePodLister")
```

Comment:
how about "pod informer not ready" error string?

Reply:
Done.